### PR TITLE
EOS-28500 - csm-agent groovy script is hardcoded for pyutils build

### PR DESCRIPTION
**Pull request by nikhilpatil2995**
# Problem Statement
- EOS-28500- csm-agent groovy script is hardcoded for pyutils build

# Design
-  For Bug, parameters are hardcoded in the cms-agent groovy script.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
http://eos-jenkins.mero.colo.seagate.com/job/GitHub-custom-ci-builds/job/centos-7.9/job/cortx-all-image-custom-ci/2184/console

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/custom-ci/centos-7.9.2009/csm-agent.groovy [hunk 1]
```diff
@@ -57,8 +57,9 @@ pipeline {
 
                 sh label: 'Configure yum repository for cortx-py-utils', script: '''
                     CORTX_UTILS_REPO_OWNER=$(echo $CORTX_UTILS_URL | cut -d "/" -f4)
-
-                    yum-config-manager --nogpgcheck --add-repo=http://cortx-storage.colo.seagate.com/releases/cortx/github/kubernetes/centos-7.9.2009/last_successful_prod/cortx_iso/
+                    
+                    yum-config-manager --add-repo=http://cortx-storage.colo.seagate.com/releases/cortx/github/integration-custom-ci/$os_version/$release_tag/cortx_iso/
+                    yum-config-manager --save --setopt=cortx-storage*.gpgcheck=1 cortx-storage* && yum-config-manager --save --setopt=cortx-storage*.gpgcheck=0 cortx-storage*
```
Comment (gauravchaudhari02) [inline jenkins/custom-ci/centos-7.9.2009/csm-agent.groovy:62]: Why are we enabling and disabling gpgcheck? we just need to disable it right?

Comment (nikhilpatil2995) [reply jenkins/custom-ci/centos-7.9.2009/csm-agent.groovy:62]: yes right. But when I have taken a reference from other script https://github.com/Seagate/cortx-re/blob/main/jenkins/custom-ci/centos-7.9.2009/hare.groovy. here we did like this

Comment (gauravchaudhari02) [reply jenkins/custom-ci/centos-7.9.2009/csm-agent.groovy:62]: @shailesh-vaidya, Any idea on this?

Comment (shailesh-vaidya) [reply jenkins/custom-ci/centos-7.9.2009/csm-agent.groovy:62]: This is due to the way `setopt` work. You need to enable it first and then disable it.

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**